File: plugins/automation/lib/discourse_automation/logger.rb
# frozen_string_literal: true

module DiscourseAutomation
  module Logger
    PREFIX = "[discourse-automation]"

    class << self
      def warn(message)
        Rails.logger.warn("#{PREFIX} #{message}")
      end

      def error(message)
        Rails.logger.error("#{PREFIX} #{message}")
      end

      def info(message)
        Rails.logger.info("#{PREFIX} #{message}")
      end
    end
  end
end

File: plugins/voice/assets/javascripts/discourse/lib/voice/logger.js
import { getOwnerWithFallback } from "discourse/lib/get-owner";
import sanitizeError from "./sanitize-error.js";

// The diagnostic message must contain only trusted metadata. Error details
// belong in the optional error argument so they pass through sanitization.
export default {
  info(message) {
    if (
      getOwnerWithFallback()?.lookup("service:site-settings")
        ?.voice_verbose_logging
    ) {
      // eslint-disable-next-line no-console
      console.info(message);
    }
  },

  warn(message, error) {
    if (
      getOwnerWithFallback()?.lookup("service:site-settings")
        ?.voice_verbose_logging
    ) {
      const details = sanitizeError(error);
      // eslint-disable-next-line no-console
      console.warn(message, ...(details ? [details] : []));
    }
  },
};
